app/jobs/fetch_oa_metadata_job.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class FetchOAMetadataJob < ApplicationJob
+  def perform(publication)
+    unpaywall_response = UnpaywallClient.query_unpaywall(publication)
+    oab_response = OABClient.query_open_access_button(publication)
+
+    if unpaywall_response.oa_locations.present?
+      publication.update_from_unpaywall(unpaywall_response)
+    elsif oab_response.url
+      publication.open_access_locations.create!(source: Source::OPEN_ACCESS_BUTTON, url: oab_response.url)
+      publication.open_access_button_last_checked_at = Time.zone.now
+    end
+
+    publication.oa_status_last_checked_at = Time.zone.now
```

Comment:
Here I think we'll also need to clear the publication's `oa_workflow_state`. It will have been set to `"oa metadata search pending"` prior to enqueuing the job, which will exclude it from any subsequent checks unless we set `oa_workflow_state` back to `nil` upon completion of the job. Does that sound right?